### PR TITLE
feat: video stream moderator

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #Adding video stream moderator

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by maze-runnar** [comment]
Don't review please.

File: app/models/video_stream_moderator.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from app.models import db
+from app.models.base import SoftDeletionModel
+
+
+class VideoStreamModerator(SoftDeletionModel):
+    __tablename__ = 'video_stream_moderator'
+
+    id = db.Column(db.Integer, primary_key=True)
+
+    event_id = db.Column(
```
Comment (iamareebjamal) [inline app/models/video_stream_moderator.py:10]: ???

Comment (maze-runnar) [reply app/models/video_stream_moderator.py:10]: I was just going to remove that. in next commit. just wanted to know what's next. Means if I am understanding the feature correctly or not.

Comment (iamareebjamal) [reply app/models/video_stream_moderator.py:10]: There is not video_stream_id here.

As discussed in the meeting, if you see meeting notes, it is written there (video_stream_id, user_id)

Now you'll have to remove all the API and schema and whatnot

That's why it was throwing error when adding relationship in the model

File: app/api/routes.py [hunk 2]
```diff
@@ -1757,7 +1757,7 @@
 api.route(
     VideoStreamModeratorDetail,
     'video_stream_moderator_detail',
-    '/video-stream-moderators/<int:id>',
+    '/video-stream-moderators/<int:video_stream_id>',
```
Comment (iamareebjamal) [inline app/api/routes.py:1760]: This should be moderator_id

File: app/api/routes.py [hunk 3]
```diff
@@ -1738,3 +1747,25 @@
 api.route(
     ExhibitorRelationship, 'exhibitor_event', '/exhibitors/<int:id>/relationships/event'
 )
+
+# VideoStreamModerator
+api.route(
+    VideoStreamModeratorList,
+    'video_stream_moderator_list',
+    '/video-streams/<int:video_stream_id>/video-stream-moderators',
```
Comment (iamareebjamal) [inline app/api/routes.py:1755]: Need /users/user_id/video-stream-moderators as well

File: app/api/schema/video_stream.py [hunk 4]
```diff
@@ -42,3 +42,12 @@ class Meta:
         schema='VideoChannelSchemaPublic',
         type_='video-channel',
     )
+    video_stream_moderator = Relationship(
```
Comment (iamareebjamal) [inline app/api/schema/video_stream.py:45]: First of all, again, this should be plural. There are multiple moderators of a video stream. It is many to many. You have written many=True as well

Secondly, the backref is moderators, so this should be `moderators`

File: app/models/video_stream_moderator.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
+from app.models import db
+from app.models.base import SoftDeletionModel
+
+
+class VideoStreamModerator(SoftDeletionModel):
```
Comment (iamareebjamal) [inline app/models/video_stream_moderator.py:5]: No soft deletion please. It is broken and doesn't work with relationships. It was shoehorned in API without much thinking

File: app/api/schema/video_stream_moderators.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
+from marshmallow_jsonapi import fields
+from marshmallow_jsonapi.flask import Relationship
+
+from app.api.helpers.utilities import dasherize
+from app.api.schema.base import SoftDeletionSchema
+
+
+class VideoStreamModeratorSchema(SoftDeletionSchema):
```
Comment (iamareebjamal) [inline app/api/schema/video_stream_moderators.py:8]: No soft deletion

File: app/api/video_stream_moderators.py [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+
+class VideoStreamModeratorList(ResourceList):
+    """
+    List and create video_stream_moderators
+    """
+
+    def query(self, view_kwargs):
+        query_ = self.session.query(VideoStreamModerator)
+        query_ = event_query(query_, view_kwargs)
```
Comment (iamareebjamal) [inline app/api/video_stream_moderators.py:17]: There is no event relation in moderator, so why event_query?

Comment (iamareebjamal) [inline app/api/video_stream_moderators.py:17]: Filter for stream_id and user_id missing

Test /video-streams/id/video-stream-moderators
/users/id/video-stream-moderators

They won't work

File: app/api/video_stream_moderators.py [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+        return query_
+
+    view_kwargs = True
+    decorators = (
+        api.has_permission(
+            'is_coorganizer', fetch='video_stream_id', model=VideoStreamModerator
```
Comment (iamareebjamal) [inline app/api/video_stream_moderators.py:24]: How can you get if a user is coorganizer from `video_stream_id`. You need event ID for that

File: app/api/video_stream_moderators.py [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    """
+    video_stream_moderators detail by id
+    """
+
+    methods = ['GET', 'PATCH', 'DELETE']
+    schema = VideoStreamModeratorSchema
+    data_layer = {
+        'session': db.session,
+        'model': VideoStreamModerator,
+    }
```
Comment (iamareebjamal) [inline app/api/video_stream_moderators.py:46]: No permissions meaning anyone can edit and delete any moderator

File: app/api/video_stream_moderators.py [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+
+
+class VideoStreamModeratorRelationship(ResourceRelationship):
+    """
+    video_stream_moderators Relationship
+    """
+
+    methods = ['GET', 'PATCH']
+    schema = VideoStreamModeratorSchema
+    data_layer = {'session': db.session, 'model': VideoStreamModerator}
```
Comment (iamareebjamal) [inline app/api/video_stream_moderators.py:56]: Same as above

File: app/models/video_stream_moderator.py [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+    __tablename__ = 'video_stream_moderators'
```
Comment (iamareebjamal) [inline app/models/video_stream_moderator.py:6]: Add unique constraints on (video_stream_id, user_id)

File: app/api/routes.py (hunk 2 above)
Comment (iamareebjamal) [reply app/api/routes.py:1760]: In fact it should be just ID

Comment (maze-runnar) [reply app/api/routes.py:1760]: it's failing tests when I am using `/video-stream-moderators/<int:id>`

File: app/models/video_stream_moderator.py [hunk 12]
```diff
@@ -0,0 +1,22 @@
+from sqlalchemy.schema import UniqueConstraint
+
+from app.models import db
+
+
+class VideoStreamModerator(db.Model):
+    __tablename__ = 'video_stream_moderators'
+    __table_args__ = (UniqueConstraint('user_id', 'video_stream_id'),)
```
Comment (iamareebjamal) [inline app/models/video_stream_moderator.py:8]: Name od constraint not given and migration not updated

**Pull request merged**